### ISSUE TITLE
Default parameters missing of maxpool2d node generated by dynamo export

### ISSUE THREAD

**Issue by leslie-fang-intel** [pytorch/pytorch#105901]
### 🐛 Describe the bug

When we use `dynamo.export` to capture the fx graph, the default parameters as input nodes for `maxpool2d` is missing. However, for `convolution`, the default parameters as input node are in the fx graph. Take the example code:
```
import torch
import torch._dynamo as torchdynamo
import copy

class ConvMaxpool2d(torch.nn.Module):
    def __init__(self,):
        super().__init__()
        self.conv = torch.nn.Conv2d(3, 64, 7, bias=True)
        self.relu = torch.nn.ReLU()
        self.maxpool = torch.nn.MaxPool2d(3, stride=2)

    def forward(self, x):
        return self.maxpool(self.relu(self.conv(x)))

def run_max_pool2d():
    batch_size = 116
    model = ConvMaxpool2d().eval()
    x = torch.randn(batch_size, 3, 224, 224).contiguous(memory_format=torch.channels_last)
    example_inputs = (x,)
    with torch.no_grad():
        # Generate the FX Module
        exported_model, guards = torchdynamo.export(
            model,
            *copy.deepcopy(example_inputs),
            aten_graph=True,
        )
        print("exported_model is: {}".format(exported_model), flush=True)

if __name__ == "__main__":
    run_max_pool2d()
```
The generated fx graph is：
```
def forward(self, x):
    arg0, = fx_pytree.tree_flatten_spec(([x], {}), self._in_spec)
    _param_constant0 = self._param_constant0
    _param_constant1 = self._param_constant1
    convolution_default = torch.ops.aten.convolution.default(arg0, _param_constant0, _param_constant1, [1, 1], [0, 0], [1, 1], False, [0, 0], 1);  arg0 = _param_constant0 = _param_constant1 = None
    relu_default = torch.ops.aten.relu.default(convolution_default);  convolution_default = None
    max_pool2d_with_indices_default = torch.ops.aten.max_pool2d_with_indices.default(relu_default, [3, 3], [2, 2]);  relu_default = None
    getitem = max_pool2d_with_indices_default[0]
    getitem_1 = max_pool2d_with_indices_default[1];  max_pool2d_with_indices_default = None
    return pytree.tree_unflatten([getitem], self._out_spec)
```

As we can see:

- For `convolution`, even we define the module as `torch.nn.Conv2d(3, 64, 7, bias=True)`, the generated graph will still take default parameters of `convolution` as input node `torch.ops.aten.convolution.default(arg0, _param_constant0, _param_constant1, [1, 1], [0, 0], [1, 1], False, [0, 0], 1);`
- However, the `maxpool2d` the generated node is ` torch.ops.aten.max_pool2d_with_indices.default(relu_default, [3, 3], [2, 2]); ` which abandons the default parameters as input nodes.
 
It will cause complicated definition in graph lowering of inductor pattern matcher. If we define the ` torch.nn.MaxPool2d(3, stride=2)` with different parameters (for example: `torch.nn.MaxPool2d(3, stride=2, padding=1)`), it will generate patterns with different number of input nodes. So, we have to pattern match each of the generated pattern.

### Versions
```
(inductor_quant) [root@CPX-4 inductor_quant]# python collect_env.py
Collecting environment information...
PyTorch version: 2.1.0a0+git9f2ad4f
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: CentOS Linux 7 (Core) (x86_64)
GCC version: (GCC) 11.2.1 20210728 (Red Hat 11.2.1-1)
Clang version: Could not collect
CMake version: version 3.22.1
Libc version: glibc-2.17

Python version: 3.8.10 (default, Jun  4 2021, 15:09:15)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-4.19.5-1.el7.elrepo.x86_64-x86_64-with-glibc2.17
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                56
On-line CPU(s) list:   0-55
Thread(s) per core:    1
Core(s) per socket:    28
Socket(s):             2
NUMA node(s):          2
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 85
Model name:            Intel Genuine CPU
Stepping:              10
CPU MHz:               1200.331
CPU max MHz:           2900.0000
CPU min MHz:           1200.0000
BogoMIPS:              5800.00
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              1024K
L3 cache:              39424K
NUMA node0 CPU(s):     0-27
NUMA node1 CPU(s):     28-55
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req pku ospke avx512_vnni flush_l1d arch_capabilities

Versions of relevant libraries:
[pip3] mypy==1.4.1
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.24.3
[pip3] torch==2.1.0a0+gitd5fc795
[pip3] torchvision==0.16.0a0+08c9938
[conda] mkl                       2023.0.0            intel_25398    intel
[conda] mkl-include               2023.0.0            intel_25398    intel
[conda] mkl-include               2023.0.0                  <pip>
[conda] mkl-service               2.4.0           py38h3605609_14    intel
[conda] mkl-static                2023.0.0                  <pip>
[conda] mkl_fft                   1.3.1           py38hcab1719_22    intel
[conda] mkl_random                1.2.2           py38hbf47bc3_22    intel
[conda] mkl_umath                 0.1.1           py38hf66a691_32    intel
[conda] numpy                     1.24.3                    <pip>
[conda] numpy                     1.22.3           py38hf0956d0_5    intel
[conda] numpy-base                1.22.3           py38h45c9ace_5    intel
```

cc @ezyang @msaroufim @wconstab @bdhirsh @anijain2305 @zou3519 @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @aakhundov

**Comment by bdhirsh** [comment]
In the example above: the [schema](https://github.com/pytorch/pytorch/blob/6f1042c0496731694635f559b9ae48a0d79d9099/aten/src/ATen/native/native_functions.yaml#L1605) of `aten.convolution`, does not have any defaultable arguments. `torch.nn.Conv2d` eventually calls `aten.conv2d`, manually filling in the arguments required for that op.

If you look at the [schema](https://github.com/pytorch/pytorch/blob/6f1042c0496731694635f559b9ae48a0d79d9099/aten/src/ATen/native/native_functions.yaml#L11746) of `max_pool2d_with_indices` though, it has some optional, defaultable arguments (e.g. `ceil_mode=None`).

For ops that have defaultable arguments, we actually work pretty hard to **not** burn the defaults into a trace, if the original code did not specify a value. This is for BC/FC: if pytorch version X+1 adds a new defaultable argument to an operator, then this scrubbing of default args allows users to serialize a model using version X+1, and load the model with a version of pytorch X. It also helps maintain BC for __torch_dispatch__ backends, who won't immediately need to update their backends if we add a defaultable kwarg to an operator.

**Comment by eellison** [comment]
Possible dupe: https://github.com/pytorch/pytorch/issues/104391

cc @aakhundov

**Comment by leslie-fang-intel** [comment]
Got it. Really thanks for the detail explanation.

**Comment by mlazos** [comment]
Resolving, feel free to reopen if other issues arise.
